src/mo_frontend/typing.ml

```diff
@@ -1914,7 +1933,13 @@ and infer_call env exp1 inst exp2 at t_expect_opt =
         error env exp2.at "M0100"
           "shared function call result contains abstract type%a"
           display_typ_expand t_ret';
-    end
+    end;
+    match T.(is_shared_sort sort || is_async t_ret'), inst.it, tbs with
+    | false, Some (true, _), ([] | T.{ sort = Type; _ } :: _) ->
+       local_error env inst.at "M0196" "unexpected `system` capability (try deleting it)"
+    | false, (None | Some (false, _)), T.{ sort = Scope; _ } :: _ ->
```

Comment:
Should we consider suppressing the warning for built-ins like cyclesAdd and setTimer? One mechansim could be adding a bool (implict|explicit) to Type. Scope (system vs @system ?) 
and only allow user-defined code to use the explicit sort and /prelude defs to access implicit. The lub would be explicit. But perhaps overkilll. (edited)

Reply:
Actually, maybe don't bother. We could always suppress the warning with a no warn directive instead.